Source/LoadByOS/WindowsConfigApp/WinInterop.cs

```diff
@@ -35,8 +36,31 @@ public void SetFolderIcon(string image, string directory)
 
         public void DeleteFolderIcon(string directory)
             => new DirectoryInfo(directory)?.DeleteIcon();
+        public bool CanUpdate => true;
+		public void InstallUpdate(string updateBundle)
+		{
+			var thisExe = Environment.ProcessPath;
+			var thisDir = Path.GetDirectoryName(thisExe);
+			var zipExtractor = Path.Combine(Path.GetTempPath(), "ZipExtractor.exe");
 
-        public void CopyTextToClipboard(string text)
-            => Clipboard.SetText(text);
+			File.Copy("ZipExtractor.exe", zipExtractor, overwrite: true);
+
+			RunAsRoot(zipExtractor, $"--input \"{updateBundle}\" --output \"{thisDir}\" --executable \"{thisExe}\"");
+		}
+
+		public Process RunAsRoot(string exe, string args)
```

Comment:
I tried to do something like this with Dinah.Core.WindowsDesktop.Processes.Runner but I think I made mine too restrictive